### PR TITLE
release bump to 0.24.0

### PR DESCRIPTION
**Pull request by tropxy**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "iso15118"
-version = "0.23.12"
+version = "0.24.0"
```
Comment (snorkman88) [inline pyproject.toml:3]: shouldn't the lock file also be updated?

Comment (tropxy) [reply pyproject.toml:3]: thanks, I am on it

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,9 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+
+## [0.24.0] - 2023-12-13
+* Big refactoring of EVSE/EVData by @tropxy in https://github.com/SwitchEV/iso15118/pull/337
```
Comment (ikaratass) [inline CHANGELOG.md:9]: for consistency, we have one extra space here.

File: pyproject.toml (hunk 1 above)
Comment (tropxy) [reply pyproject.toml:3]: done

File: CHANGELOG.md (hunk 2 above)
Comment (tropxy) [reply CHANGELOG.md:9]: good catch, thanks

**Review by shalinnijel2 — APPROVED** [review]

**Review by snorkman88 — APPROVED** [review]

**Pull request merged**